node/src/manager/commands/poi.rs

```diff
@@ -0,0 +1,655 @@
+
+    return Ok(json.divergent_blocks);
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct CallCacheRecord {
+    #[sql_type = "Text"]
+    id: String,
+    #[sql_type = "Text"]
+    return_value: String,
+    #[sql_type = "Text"]
+    contract_address: String,
+    #[sql_type = "Integer"]
+    block_number: i32,
+}
+
+pub fn get_call_cache(
+    conn: &PooledConnection<ConnectionManager<PgConnection>>,
+    chain_namespace: &str,
+    divergent_block: i32,
+) -> Result<Vec<CallCacheRecord>, anyhow::Error> {
+    let raw_query = format!(
+        "SELECT encode(id,'hex') as id, encode(return_value,'hex') as \
+        return_value,encode(contract_address,'hex') as contract_address,\
+        block_number from {}.call_cache WHERE block_number = {}",
+        chain_namespace, divergent_block
+    );
+
+    let callcache_queryset = sql_query(raw_query).load::<CallCacheRecord>(conn).unwrap();
+    Ok(callcache_queryset)
+}
+
+fn dump_call_cache(
+    directory: &PathBuf,
+    callcache_records: std::vec::Vec<CallCacheRecord>,
+) -> Result<(), anyhow::Error> {
+    println!("dumping call cache");
+    let filepath = directory.join("call_cache.tsv");
+
+    let mut csv_writer_callcache = WriterBuilder::new().delimiter(b'\t').from_path(filepath)?;
+
+    // let f = File::create(&filepath).expect("Unable to create file");
+    // let mut file_buffer = BufWriter::new(f);
+    csv_writer_callcache.write_record(&[
+        "id",
+        "return_value",
+        "contract_address",
+        "block_number",
+    ])?;
+
+    for cc in &callcache_records {
+        csv_writer_callcache.write_record(&[
+            &cc.id,
+            &cc.return_value,
+            &cc.contract_address,
+            &cc.block_number.to_string(),
+        ])?;
+    }
+    csv_writer_callcache.flush()?;
+    return Ok(());
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct TableName {
+    #[sql_type = "Text"]
+    tablename: String,
+}
+
+/// Get all tables present in the subgraph schema. Will be used for dumping data
+fn get_subgraph_tables(
+    conn: &PooledConnection<ConnectionManager<PgConnection>>,
+    subgraph_schema_name: &str,
+) -> Result<Vec<TableName>, anyhow::Error> {
+    let query = format!(
+        "select tablename from pg_tables where schemaname='{}'",
+        subgraph_schema_name
+    );
+
+    let subgraph_tables = sql_query(query).get_results::<TableName>(conn).unwrap();
+    return Ok(subgraph_tables);
+}
+
+#[derive(Queryable, QueryableByName)]
+pub struct ColumnNameAndType {
+    #[sql_type = "Text"]
+    column_name: String,
+    #[sql_type = "Text"]
+    data_type: String,
+}
+
+/// Gathers entries from entity tables that contain the divergent block if in closed interval
+/// Diesel isn't of any use here because this is dynamic.
+/// Also need to catch the columns for passing to csv.
+///
+/// Note: there may be no return values for a table filtered on a block. In this case, eagerly return
+/// Note: map null values to empty string;
```

Comment:
This replicates a lot of the work that things like the `Layout` and `relational_queries` in the store do already and in a way that's very dependent on the details of how data is stored which shouldn't leak outside of the `store` crate.

I would much prefer if we added a new variant to `EntityFilter`, say `ChangedAtBlock(BlockNumber)` so that you can just use `QueryStore::find_query_values` or `SubgraphStore::find` to get the entities you want. Having an official way to get the entities that were modified at a specific block would be generally useful.

You can find all the entity types by getting the subgraph's schema with `SubgraphStore::input_schema` and then using `DocumentExt::get_object_type_definitions()` on it. You'd then construct one `EntityQuery` for each object type and get back a list of entities.